tests/e2e/util.go

```diff
@@ -3339,47 +3338,54 @@ func toggleCSIMigrationFeatureGatesOnKubeControllerManager(ctx context.Context,
-		User: "root",
-		Auth: []ssh.AuthMethod{
-			ssh.Password("ca$hc0w"),
-		},
-		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
-	}
-	result, err := sshExec(sshClientConfig, k8sMasterIP, grepCmd)
-	if err != nil {
-		return err
-	}
-	if err != nil {
-		fssh.LogResult(result)
-		return fmt.Errorf("command failed/couldn't execute command: %s on host: %v , error: %s",
-			grepCmd, k8sMasterIP, err)
-	}
-	if result.Code != 0 {
-		if add {
-			// nolint:misspell
-			sshCmd = "gawk -i inplace '/--bind-addres/ " +
-				"{ print; print \"    - --feature-gates=CSIMigration=true,CSIMigrationvSphere=true\"; next }1' " +
-				kcmManifest
-		} else {
-			return nil
+	k8sMasterIPs := getK8sMasterIPs(ctx, client)
+	for _, k8sMasterIP := range k8sMasterIPs {
+		framework.Logf("Invoking command '%v' on host %v", grepCmd, k8sMasterIP)
+		sshClientConfig := &ssh.ClientConfig{
+			User: "root",
+			Auth: []ssh.AuthMethod{
+				ssh.Password(esxPassword),
+			},
+			HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+		}
+		result, err := sshExec(sshClientConfig, k8sMasterIP, grepCmd)
+		if err != nil {
+			return err
+		}
+		if err != nil {
+			fssh.LogResult(result)
+			return fmt.Errorf("command failed/couldn't execute command: %s on host: %v , error: %s",
+				grepCmd, k8sMasterIP, err)
+		}
+		if result.Code != 0 {
+			if add {
+				// nolint:misspell
+				sshCmd = "gawk -i inplace '/--bind-addres/ " +
+					"{ print; print \"    - --feature-gates=CSIMigration=true,CSIMigrationvSphere=true\"; next }1' " +
+					kcmManifest
+			} else {
+				return nil
+			}
+		}
+		framework.Logf("Invoking command %v on host %v", sshCmd, k8sMasterIP)
+		result, err = sshExec(sshClientConfig, k8sMasterIP, sshCmd)
+		if err != nil || result.Code != 0 {
+			fssh.LogResult(result)
+			return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s", sshCmd, k8sMasterIP, err)
 		}
-	}
-	framework.Logf("Invoking command %v on host %v", sshCmd, k8sMasterIP)
-	result, err = sshExec(sshClientConfig, k8sMasterIP, sshCmd)
-	if err != nil || result.Code != 0 {
-		fssh.LogResult(result)
-		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s", sshCmd, k8sMasterIP, err)
 	}
 	// Sleeping for two seconds so that the change made to manifest file is
 	// recognised.
 	time.Sleep(2 * time.Second)
-	framework.Logf("Waiting for 'kube-controller-manager' controller pod to come up within %v seconds", pollTimeout)
+
+	framework.Logf(
+		"Waiting for 'kube-controller-manager' controller pod to come up within %v seconds", pollTimeout*2)
 	label := labels.SelectorFromSet(labels.Set(map[string]string{"component": "kube-controller-manager"}))
-	_, err = fpod.WaitForPodsWithLabelRunningReady(client, kubeSystemNamespace, label, 1, pollTimeout)
-	framework.Logf("'kube-controller-manager' controller pod is up and ready within %v seconds", pollTimeout)
+	_, err = fpod.WaitForPodsWithLabelRunningReady(
+		client, kubeSystemNamespace, label, len(k8sMasterIPs), pollTimeout*2)
+	if err == nil {
```

Comment:
Better to log when there is an error instead of non-error

Reply:
Done